### DISCUSSION TITLE
Create new sheets within a spreadsheet

### DISCUSSION THREAD

**Discussion by DarkTherapy** [mobizt/ESP-Google-Sheet-Client#20] [Q&A]
HI, can this library create new sheets within a spreadsheet? sheet1, sheet2, sheet3 etc..

I cant seem to find any info about creating them with this library.  I need to be able to add multiple sheets remotely with default column names.

**Comment by mobizt** [comment]
The API supports for single spreadsheet creation.
https://github.com/mobizt/ESP-Google-Sheet-Client/blob/master/examples/Spreadsheets/Create/Create.ino

You should keep in mind the different between document and data. 
You can create and modify data anytime you want but the document stored as object in the storage which takes more resource and time for operation.

No cloud services provide such your requirement.

**Reply by DarkTherapy** [reply]
OK, so the library can only create new spreadsheets for my data.. but cant make new sheets to send my data to.

**Comment by mobizt** [comment]
Ok I get your point, to create multiple sheets, you should read the documentation and try it because it can be complex.
https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets

For example, you can try.

```cpp
FirebaseJson spreadsheet;
spreadsheet.set("properties/title", "spread sheet 1");
spreadsheet.set("sheets/[0]/properties/title", "Sheet1");
spreadsheet.set("sheets/[0]/properties/sheetId", 1);
spreadsheet.set("sheets/[0]/properties/sheetType", "GRID");
spreadsheet.set("sheets/[0]/properties/sheetType", "GRID");
spreadsheet.set("sheets/[0]/properties/gridProperties/rowCount", 20);
spreadsheet.set("sheets/[0]/properties/gridProperties/columnCount", 10);

spreadsheet.set("sheets/[0]/developerMetadata/[0]/metadataValue", "test_value");
spreadsheet.set("sheets/[0]/developerMetadata/[0]/metadataKey", "test_yey");
spreadsheet.set("sheets/[0]/developerMetadata/[0]/visibility", "DOCUMENT");

spreadsheet.set("sheets/[1]/properties/title", "Sheet2");
spreadsheet.set("sheets/[1]/properties/sheetId", 1);
spreadsheet.set("sheets/[1]/properties/sheetType", "GRID");
spreadsheet.set("sheets/[1]/properties/sheetType", "GRID");
spreadsheet.set("sheets/[1]/properties/gridProperties/rowCount", 20);
spreadsheet.set("sheets/[1]/properties/gridProperties/columnCount", 10);

spreadsheet.set("sheets/[1]/developerMetadata/[0]/metadataValue", "test_value");
spreadsheet.set("sheets/[1]/developerMetadata/[0]/metadataKey", "test_yey");
spreadsheet.set("sheets/[1]/developerMetadata/[0]/visibility", "DOCUMENT");
```

Library works as a REST API client, and you should read the Google documentation and try it yourself.

**Reply by DarkTherapy** [reply]
Thank you
